pkg/scheduler/profile/profile.go

```diff
@@ -86,20 +85,25 @@ type cfgValidator struct {
 	queueSortArgs runtime.Object
 }
 
-func (v *cfgValidator) validate(cfg config.KubeSchedulerProfile) error {
-	if len(cfg.SchedulerName) == 0 {
+func (v *cfgValidator) validate(cfg config.KubeSchedulerProfile, f framework.Framework) error {
+	if len(f.ProfileName()) == 0 {
 		return errors.New("scheduler name is needed")
 	}
 	if cfg.Plugins == nil {
-		return fmt.Errorf("plugins required for profile with scheduler name %q", cfg.SchedulerName)
+		return fmt.Errorf("plugins required for profile with scheduler name %q", f.ProfileName())
+	}
+	if v.m[f.ProfileName()] != nil {
+		return fmt.Errorf("duplicate profile with scheduler name %q", f.ProfileName())
 	}
-	if v.m[cfg.SchedulerName] != nil {
-		return fmt.Errorf("duplicate profile with scheduler name %q", cfg.SchedulerName)
+	if len(f.ListPlugins().QueueSort.Enabled) != 1 {
+		return fmt.Errorf("one queue sort plugin required for profile with scheduler name %q", f.ProfileName())
 	}
-	if len(cfg.Plugins.QueueSort.Enabled) != 1 {
-		return fmt.Errorf("one queue sort plugin required for profile with scheduler name %q", cfg.SchedulerName)
+
+	queueSort := f.ListPlugins().QueueSort.Enabled[0].Name
+	if len(queueSort) == 0 {
```

Comment:
tl;dr: I fixed the test case to preserve the plugin names

It's actually a conflict of two different tests based on what functions they involve. If we set this line to use queueSort from the config (ie, `queueSort := cfg.Plugins.QueueSort.Enabled[0].Name`), then `TestSetup` fails. This failure is because the default plugins now come from MultiPoint, so there are no `QueueSort` plugins in the config until they are expanded by the factory logic:
```
$ go test ./cmd/kube-scheduler/...
...
--- FAIL: TestSetup (0.78s)
    --- FAIL: TestSetup/default_config (0.78s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 100 [running]:
testing.tRunner.func1.2({0x2819100, 0xc00031e150})
	/usr/local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1212 +0x218
panic({0x2819100, 0xc00031e150})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
k8s.io/kubernetes/pkg/scheduler/profile.(*cfgValidator).validate(0xc000a6b670, {{0x294deb8, 0x11}, 0xc0004b38c0, {0xc000156540, 0x7, 0x7}}, {0x2c37768, 0xc00025c8c0})
	/Users/mdame/go/src/k8s.io/kubernetes/pkg/scheduler/profile/profile.go:103 +0x5b0
k8s.io/kubernetes/pkg/scheduler/profile.NewMap({0xc00044b8f0, 0x1, 0x0}, 0xffffffffffffffff, 0x0, {0xc000a6b9b8, 0xb, 0xb})
	/Users/mdame/go/src/k8s.io/kubernetes/pkg/scheduler/profile/profile.go:61 +0x1d9
k8s.io/kubernetes/pkg/scheduler.(*Configurator).create(0xc000a6bae0)
	/Users/mdame/go/src/k8s.io/kubernetes/pkg/scheduler/factory.go:143 +0xbd3
k8s.io/kubernetes/pkg/scheduler.New({0x2c396d0, 0xc0001d4f00}, {0x2c2e848, 0xc00076c140}, {0x2bd9ad0, 0xc00076c1e0}, 0xc00052c600, 0x0, {0xc000907d20, 0xa, ...})
	/Users/mdame/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:264 +0x47b
k8s.io/kubernetes/cmd/kube-scheduler/app.Setup({0x2be0f98, 0xc00004eec0}, 0xc0003c0780, {0x0, 0x0, 0x3b7dbe0})
	/Users/mdame/go/src/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go:309 +0x5d8
k8s.io/kubernetes/cmd/kube-scheduler/app.TestSetup.func3(0xc000582b60)
	/Users/mdame/go/src/k8s.io/kubernetes/cmd/kube-scheduler/app/server_test.go:241 +0x347
testing.tRunner(0xc000582b60, 0xc0006c4678)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x35a
...
```

But, if we flip it and just check the plugins from the completed factory (ie, `queueSort := f.ListPlugins().QueueSort.Enabled[0].Name`), then `TestNewMap` fails because we don't get the expected error:
```
$ go test ./pkg/scheduler/...
...
--- FAIL: TestNewMap (0.00s)
    --- FAIL: TestNewMap/different_queue_sort (0.00s)
        profile_test.go:251: got error %!q(<nil>), want "different queue sort plugins"
    --- FAIL: TestNewMap/different_queue_sort_args (0.00s)
        profile_test.go:251: got error %!q(<nil>), want "different queue sort plugin args"
FAIL
FAIL	k8s.io/kubernetes/pkg/scheduler/profile	0.345s
```

This is the test that is causing the issue. I debugged this to find that `f.ListPlugins()` was returning a list of plugins, but they didn't have any registered names. So, looking at `updatePluginList` in `framework.go`, I realized that we are just appending the plugin object. I'm assuming we then use `p.Name()` later on, because all of these test plugins are registered with the same empty `Name()` function. 

I fixed this by adding a wrapper to the test cases to pass an actual plugin name to each plugin. We might want to strengthen the checks for this if possible rather than just putting a bandaid on it for the test itself.